var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -62,12 +82,37 @@ def install(self, spec, prefix):
                 '--with-hwloc=%s' % spec['hwloc'].prefix,
                 '--with-serial'
             ]
+            arch_args = []
+            host_arch_args = []
+            gpu_arch_args = []
+            # backends
             if '+openmp' in spec:
                 g_args.append('--with-openmp')
             if 'qthreads' in spec:
                 g_args.append('--with-qthreads=%s' % spec['qthreads'].prefix)
             if 'cuda' in spec:
                 g_args.append('--with-cuda=%s' % spec['cuda'].prefix)
+            # host architectures
+            host_arch_args = spec.variants['host_arch'].value
+
+            # gpu architectures
+            gpu_arch_args  = spec.variants['gpu_arch'].value
+
+            # only a host architecture
+            if (host_arch_args != 'none' AND gpu_arch_args == ''):
+                arch_args = '--arch=' + host_arch_args
+            # only a gpu architecture
+            if (host_arch_args == '' AND gpu_arch_args != 'none'):
+                if '+cuda' in spec:
+                    arch_args = '--arch=' + gpu_arch_args
+            # both a host and a gpu architecture
+            if (host_arch_args != 'none' AND gpu_arch_args != 'none'):
+                if '+cuda' in spec:
+                    arch_args = '--arch=' + host_arch_args
+                    + ',' + gpu_arch_args
+
+            if arch_args != '':
+                g_args += arch_args
```

Comment:
You initialize `arch_flags` as `[]`, not `''`, so I'm not sure if this check actually works. Also, shouldn't you be appending? `+=` doesn't work the way you think:
```python
>>> x = []
>>> x += 'foo'
>>> x
['f', 'o', 'o']
```